xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/user/impl/xwiki/UserAuthenticatedManager.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package com.xpn.xwiki.user.impl.xwiki;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.xpn.xwiki.web.Utils;
+
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.security.authentication.UserAuthenticatedEvent;
+import org.xwiki.stability.Unstable;
+import org.xwiki.user.UserReference;
+import org.xwiki.user.UserReferenceResolver;
+
+/**
+ * Wraps an {@link org.xwiki.observation.ObservationManager} and a {@link org.xwiki.user.UserReferenceResolver} to
+ * notify about user who are authenticated through {@link com.xpn.xwiki.user.impl.xwiki.MyFormAuthenticator} and {@link
+ * com.xpn.xwiki.user.impl.xwiki.MyBasicAuthenticator}.
+ *
+ * @version $Id$
+ * @since 13.3RC1
+ */
+@Unstable
+public class UserAuthenticatedManager
```

Comment:
I really don't think we want to make that public right now, so I would move it in an internal package (for example `com.xpn.xwiki.internal.user`). I would also make it a component instead of a simple Java object so that it can use injection (see https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java#L99 for a common example of a utility internal component declaring itself as role to avoid useless plumbing for something which is internal).